### PR TITLE
chore(fail steps): Update failsteps for GCP, Kafka, Cassandra, and Springboot Experiments

### PR DESCRIPTION
**Pull request by neelanjan00**
**What this PR does / why we need it**:
- Update fail steps for GCP, Kafka, Cassandra, and Springboot Experiments

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
-   [ ] Fixes #<issue number>
-   [ ] PR messages has document related information
-   [ ] Labelled this PR & related issue with `breaking-changes` tag
-   [ ] PR messages has breaking changes related information
-   [ ] Labelled this PR & related issue with `requires-upgrade` tag
-   [ ] PR messages has upgrade related information
-   [ ] Commit has unit tests
-   [ ] Commit has integration tests
-   [ ] E2E run Required for the changes

### CODE REVIEW

File: chaoslib/litmus/gcp-vm-disk-loss-by-label/lib/gcp-vm-disk-loss-by-label.go [hunk 1]
```diff
@@ -62,14 +64,14 @@ func PrepareDiskVolumeLossByLabel(computeService *compute.Service, experimentsDe
...
+				return stacktrace.Propagate(err, "could not run chaos in serial mode")
 			}
 		case "parallel":
 			if err = injectChaosInParallelMode(computeService, experimentsDetails, diskVolumeNamesList, experimentsDetails.TargetDiskInstanceNamesList, experimentsDetails.Zones, clients, resultDetails, eventsDetails, chaosDetails); err != nil {
-				return err
+				return stacktrace.Propagate(err, "could not run chaos in parallel mode")
 			}
 		default:
-			return errors.Errorf("%v sequence is not supported", experimentsDetails.Sequence)
+			return cerrors.Error{Phase: "ChaosInject", Reason: fmt.Sprintf("'%s' sequence is not supported", experimentsDetails.Sequence)}
```
Comment (ispeakc0de) [inline chaoslib/litmus/gcp-vm-disk-loss-by-label/lib/gcp-vm-disk-loss-by-label.go:74]: we can use a generic error code for this as it is not a chaos inject step

File: chaoslib/litmus/gcp-vm-disk-loss-by-label/lib/gcp-vm-disk-loss-by-label.go [hunk 2]
```diff
@@ -102,15 +104,15 @@ func injectChaosInSerialMode(computeService *compute.Service, experimentsDetails
 			//Detaching the disk volume from the instance
 			log.Info("[Chaos]: Detaching the disk volume from the instance")
 			if err = gcp.DiskVolumeDetach(computeService, instanceNamesList[i], experimentsDetails.GCPProjectID, zone, experimentsDetails.DeviceNamesList[i]); err != nil {
-				return errors.Errorf("disk detachment failed, err: %v", err)
+				return stacktrace.Propagate(err, "disk detachment failed")
```
Comment (ispeakc0de) [inline chaoslib/litmus/gcp-vm-disk-loss-by-label/lib/gcp-vm-disk-loss-by-label.go:107]:
```suggestion
				return stacktrace.Propagate(err, "could not detach disk")
```

Comment (ispeakc0de) [reply chaoslib/litmus/gcp-vm-disk-loss-by-label/lib/gcp-vm-disk-loss-by-label.go:107]: let's use the same wrapper msg as much as possible

File: chaoslib/litmus/kafka-broker-pod-failure/lib/pod-delete.go [hunk 3]
```diff
@@ -109,13 +111,13 @@ func injectChaosInSerialMode(experimentsDetails *experimentTypes.ExperimentDetai
 				err = clients.KubeClient.CoreV1().Pods(pod.Namespace).Delete(context.Background(), pod.Name, v1.DeleteOptions{})
 			}
 			if err != nil {
-				return err
+				return cerrors.Error{ErrorCode: cerrors.ErrorTypeGeneric, Target: fmt.Sprintf("{podName: %s, namespace: %s}", pod.Name, pod.Namespace), Reason: fmt.Sprintf("failed to delete the target pod: %s", err.Error())}
```
Comment (ispeakc0de) [inline chaoslib/litmus/kafka-broker-pod-failure/lib/pod-delete.go:114]: Use ChaosInject errorcode here

File: chaoslib/litmus/spring-boot-chaos/lib/spring-boot-chaos.go [hunk 4]
```diff
@@ -110,7 +112,7 @@ func CheckChaosMonkey(chaosMonkeyPort string, chaosMonkeyPath string, targetPods
 	}
 
 	if hasErrors {
-		return false, errors.Errorf("failed to check chaos moonkey on at least one pod, check logs for details")
+		return false, cerrors.Error{ErrorCode: cerrors.ErrorTypeStatusChecks, Reason: "failed to check chaos monkey on at least one pod, check logs for details"}
```
Comment (ispeakc0de) [inline chaoslib/litmus/spring-boot-chaos/lib/spring-boot-chaos.go:115]: add Targets

Comment (ispeakc0de) [reply chaoslib/litmus/spring-boot-chaos/lib/spring-boot-chaos.go:115]: check other places too - also remove targets from reason and add them separately inside target field

File: pkg/cassandra/liveness.go [hunk 5]
```diff
@@ -61,29 +63,29 @@ func LivenessCleanup(experimentsDetails *experimentTypes.ExperimentDetails, clie
 	log.Info("[CleanUP]: Getting ClusterIP of liveness service")
 	ClusterIP, err := GetServiceClusterIP(experimentsDetails, clients)
 	if err != nil {
-		return errors.Errorf("failed to get the ClusterIP of liveness service, err: %v", err)
+		return cerrors.Error{ErrorCode: cerrors.ErrorTypeChaosRevert, Reason: fmt.Sprintf("failed to get the ClusterIP of liveness service, err: %v", err)}
```
Comment (ispeakc0de) [inline pkg/cassandra/liveness.go:66]: Use err.Error() inside of err inside reason

File: pkg/cassandra/liveness.go [hunk 6]
```diff
@@ -61,29 +63,29 @@ func LivenessCleanup(experimentsDetails *experimentTypes.ExperimentDetails, clie
...
 
 	if err = WaitTillCycleComplete(experimentsDetails, ClusterIP); err != nil {
-		return errors.Errorf("cycle complete test failed, err: %v", err)
+		return cerrors.Error{ErrorCode: cerrors.ErrorTypeChaosRevert, Reason: fmt.Sprintf("cycle complete test failed, err: %v", err)}
 	}
 
 	log.Info("[Cleanup]: Deleting cassandra liveness deployment & service")
 	if err = DeleteLivenessDeployment(experimentsDetails, clients); err != nil {
-		return errors.Errorf("liveness deployment deletion failed, err: %v", err)
+		return cerrors.Error{ErrorCode: cerrors.ErrorTypeChaosRevert, Reason: fmt.Sprintf("liveness deployment deletion failed, err: %v", err)}
```
Comment (ispeakc0de) [inline pkg/cassandra/liveness.go:85]: It is Generic error as this is not related to chaos - it is just to check the liveness of target application

File: pkg/cassandra/liveness.go [hunk 7]
```diff
@@ -366,7 +368,7 @@ func CreateLivenessService(experimentsDetails *experimentTypes.ExperimentDetails
 	// Creating liveness service
 	_, err := clients.KubeClient.CoreV1().Services(experimentsDetails.ChaoslibDetail.AppNS).Create(context.Background(), livenessSvc, metav1.CreateOptions{})
 	if err != nil {
-		return err
+		return cerrors.Error{ErrorCode: cerrors.ErrorTypeStatusChecks, Reason: fmt.Sprintf("unable to create liveness service, err: %v", err)}
```
Comment (ispeakc0de) [inline pkg/cassandra/liveness.go:371]: add target

File: pkg/cloud/gcp/get-credentials-json.go [hunk 8]
```diff
@@ -47,60 +49,60 @@ func getServiceAccountJSONFromSecret() ([]byte, error) {
...
 	gcpType, err := getFileContent("/tmp/type")
 	if err != nil {
-		return nil, err
+		return nil, cerrors.Error{ErrorCode: cerrors.ErrorTypeGeneric, Reason: fmt.Sprintf("failed to read file /tmp/type, err: %v", err)}
 	}
 
 	gcpProjectID, err := getFileContent("/tmp/project_id")
 	if err != nil {
-		return nil, err
+		return nil, cerrors.Error{ErrorCode: cerrors.ErrorTypeGeneric, Reason: fmt.Sprintf("failed to read file /tmp/project_id, err: %v", err)}
```
Comment (ispeakc0de) [inline pkg/cloud/gcp/get-credentials-json.go:57]: use err.Error()

Comment (ispeakc0de) [inline pkg/cloud/gcp/get-credentials-json.go:57]: getFileContent already returning a user-friendly error. we can skip this

File: pkg/cloud/gcp/get-credentials-json.go [hunk 9]
```diff
@@ -30,7 +32,7 @@ func getFileContent(filePath string) (string, error) {
 
 	fileContentByteSlice, err := os.ReadFile(filePath)
 	if err != nil {
-		return "", err
+		return "", cerrors.Error{ErrorCode: cerrors.ErrorTypeGeneric, Reason: fmt.Sprintf("failed to read file %v, err: %v", filePath, err)}
```
Comment (ispeakc0de) [inline pkg/cloud/gcp/get-credentials-json.go:35]: use err.Error()

File: pkg/kafka/kafka-liveness-cleanup.go [hunk 10]
```diff
@@ -2,20 +2,21 @@ package kafka
...
-	"github.com/pkg/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // LivenessCleanup deletes the kafka liveness pod
 func LivenessCleanup(experimentsDetails *experimentTypes.ExperimentDetails, clients clients.ClientSets) error {
 
 	if err := clients.KubeClient.CoreV1().Pods(experimentsDetails.ChaoslibDetail.AppNS).Delete(context.Background(), "kafka-liveness-"+experimentsDetails.RunID, metav1.DeleteOptions{}); err != nil {
-		return errors.Errorf("Fail to delete liveness deployment, err: %v", err)
+		return cerrors.Error{ErrorCode: cerrors.ErrorTypeChaosRevert, Reason: fmt.Sprintf("fail to delete liveness deployment, err: %v", err)}
```
Comment (ispeakc0de) [inline pkg/kafka/kafka-liveness-cleanup.go:19]: use Generic

File: chaoslib/litmus/gcp-vm-disk-loss-by-label/lib/gcp-vm-disk-loss-by-label.go (hunk 2 above)
Comment (neelanjan00) [reply chaoslib/litmus/gcp-vm-disk-loss-by-label/lib/gcp-vm-disk-loss-by-label.go:107]: Will this be necessary? Then we would have to make changes in many places

File: pkg/cassandra/liveness.go (hunk 7 above)
Comment (neelanjan00) [reply pkg/cassandra/liveness.go:371]: But we don't have the target pod details here... We are filtering it as part of the generic/pod-delete lib logic

Comment (ispeakc0de) [reply pkg/cassandra/liveness.go]: Here Target is liveness service name
